tests/consumer/test_oauth1.py

```diff
@@ -323,6 +323,22 @@ def callback(*args, **kwargs):
     assert len(calls) == 1
 
 
+@requires_blinker
+def test_signal_oauth_before_login(request):
+    app, bp = make_app()
+    bp.session.fetch_request_token = mock.Mock(return_value="test-token")
+    def callback(*args, **kwargs):
+        flask.session["test"] = "test"
```

Comment:
Nitpick: can you make this test do something a little more like a real-world use-case?

Reply:
I use it to log the user out before the login. Here's why:
I am using Flask as an API service, so the flow is:
User clicks on "login with X (facebook, instagram etc.)" from mobile --> browser opens with the server. Now here's the gotcha - I need to know if the user is authenticated - so the server can decide whether it needs to associate a new network for a specific user, or create a new user. I need a login session opened and closed for each request, but before opening a new session - I want to clear any unwanted ones that may exist - and I do it with `before_login`.

Reply:
What I'm trying to say is that I think that saving a session before the redirect is indeed a real world use case.

Reply:
Sure. Then can you do something like “flask.session[“google_id”] = 123” before the code being tested, then do “del flask.session[“google_id”]” in the signal handler, then assert that there is no “google_id” in the Flask session at the end of the test?

Reply:
Done.